pkg/awsutil/secretsmgr/api.go

```diff
@@ -0,0 +1,19 @@
+package secretsmgr
+
+import (
+	"time"
+
+	"github.com/Cloud-Foundations/golib/pkg/log"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+)
+
+func GetAwsSecret(metadataClient *ec2metadata.EC2Metadata,
```

Comment:
These apis' feel clunky. I Thikn you should define a new type and the call to new specify the duration of the cache.

Reply:
Hm. If I create a type for it, perhaps it's cleaner to drop the global cache and instead create a type which is a handle to the specific secret?

Reply:
Changes made. PTAL before I push more into this PR :-)